src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunStandalongMethodMetadata.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+using Internal.IL;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+
+namespace ILCompiler
+{
+    // Alternate form of metadata that represents a single method. Self contained except for type references
+    public class ReadyToRunStandaloneMethodMetadata
+    {
+        public byte[] ConstantData;
+        public TypeDesc[] TypeRefs;
+
+        public static ReadyToRunStandaloneMethodMetadata Compute(EcmaMethod wrappedMethod)
+        {
+            var metadataReader = wrappedMethod.MetadataReader;
+            var _module = wrappedMethod.Module;
+            var rva = wrappedMethod.MetadataReader.GetMethodDefinition(wrappedMethod.Handle).RelativeVirtualAddress;
+            var _methodBody = _module.PEReader.GetMethodBody(rva);
+
+            byte[] _alternateILStream = _methodBody.GetILBytes();
+
+            var exceptionRegions = _methodBody.ExceptionRegions;
+            ILExceptionRegion[] _exceptionRegions;
+
+            int length = exceptionRegions.Length;
+            if (length == 0)
+            {
+                _exceptionRegions = Array.Empty<ILExceptionRegion>();
+            }
+            else
+            {
+                _exceptionRegions = new ILExceptionRegion[length];
+                for (int i = 0; i < length; i++)
+                {
+                    var exceptionRegion = exceptionRegions[i];
+
+                    _exceptionRegions[i] = new ILExceptionRegion(
+                        (ILExceptionRegionKind)exceptionRegion.Kind, // assumes that ILExceptionRegionKind and ExceptionRegionKind enums are in sync
```

Comment:
Eh, there is quite a lot of dependence on these being identical, also there is the same comment in the EcmaMethodILProvider as well. This will be fine.